README.md

```diff
@@ -1847,6 +1847,8 @@ ANSWERS.addComponent('Map', {
   // Optional, determines if an empty map should be shown when there are no results. Defaults to false.
   showEmptyMap: false,
   // Optional, callback to invoke when a pin is clicked. The clicked item(s) are passed to the callback
+  shouldFreezeMap: function() {},
```

Comment:
Why is this a function instead of a boolean?

Reply:
I see. This is dynamic for that mobile vs desktop case.

Reply:
It is odd to me that we allow this to be a function when a lot of our other boolean config flags are not also functions. It's almost like we are catering this boolean flag specifically for the mobile vs. desktop case. It makes me think we would want a more general way to do mobile vs. desktop so that it can apply to more than just this config option.

Reply:
yeah let me think about this for a bit, I think we also want some config option to support disabling the map when the page is displaying a full screen modal, but currently that wouldn't work because closing the modal wouldn't "unfreeze" the map

Reply:
Actually there's already many ways to "unfreeze" the map. This could be done by capturing a reference to the map component with something like the below

```js
let map;
ANSWERS.addComponent('Map', {
    container: '.map-container-google',
    onCreate: function () {
        map = this;
    }
});
```

or maybe registering an additional global storage with something like

```js
ANSWERS.addComponent('Map', {
    container: '.map-container-google',
    onCreate: function () {
        this.core.globalStorage.on('update', 'some-custom-key', () => this.setState(this.getState()));
    }
});
```

or you could use the ANSWERS.components instance to get a reference to the component with a 

```js
ANSWERS.components.getActiveComponent('Map');
```

Anyways that was all really extra, but right now if somebody wanted to do something like a full screen modal, there's no public way to pause the map. I think the easiest way would be to add an onMountOverride, that has logic around the this._map, or completely override the component and copy in/rewrite the map providers.

Does that make this sound slightly less scoped to my specific use case? I'm not sure if it does, but the only other way forward I see would be to export the SDKs map providers, and write a custom map component for them.

Reply:
or tell Jeremy he has to be happy with poor mobile performance

Reply:
I still get the feeling that we want static configuration when we can. I feel /less bad/ about some `ANSWERS.setMapFreeze` but as we talked about online, it's weird that answers would know about the internals of the Map component.

Reply:
As I understand it, we want to have dynamic map freezing/unfreezing on the page (for the case of desktop/mobile and an overlay, to better performance). The reason I suggested above is because the only other cases we have this is for like `ANSWERS.setContext` or `ANSWERS.setGeolocation`. But if want to separate UI from Core Search that is not great.

Reply:
tl;dr I am not sure the way forward, my gut tells me if we expose a function here we give precedence to expose a function in a lot of other places

Reply:
@tmeyer2115 do you have an opinion on this? I was thinking if we don't like this function I could expose the map providers instead, and write a custom map component for the new page template

Reply:
Tom's suggestion of only adding the map when not mobile makes a lot of sense! I previously thought we would have to rely on the screen width to do this, which would cause issues if somebody resized their screen (say from maximizing it or something) BUT we can actually do that with navigator.useragent

Reply:
I did this by forking a tiny method https://github.com/juliangruber/is-mobile/blob/master/index.js